packages/common/__tests__/utils/nft.utils.test.ts

```diff
@@ -320,3 +473,503 @@ describe('minor helpers', () => {
+            token: tokenIndex < 0 ? hathorLib.constants.HATHOR_TOKEN_CONFIG.uid : fullNodeData.tokens[tokenIndex],
+            value: input.spent_output.value,
+            token_data: input.spent_output.token_data,
+            script: input.spent_output.script,
+            decoded: {
+              ...input.spent_output.decoded,
+            },
+            tx_id: input.tx_id,
+            index: input.index
+          };
+        }),
+        outputs: fullNodeData.outputs.map((output) => {
+          const tokenIndex = (output.token_data & hathorLib.constants.TOKEN_INDEX_MASK) - 1;
+          return {
+            ...output,
+            decoded: output.decoded ? output.decoded : {},
+            spent_by: null,
+            token: tokenIndex < 0 ? hathorLib.constants.HATHOR_TOKEN_CONFIG.uid : fullNodeData.tokens[tokenIndex],
+          };
+        }),
+      };
+
+      // Test that the transformed transaction is valid for NFT handling
+      const network = { name: 'testnet' };
+      const result = NftUtils.shouldInvokeNftHandlerForTx(txFromEvent, network as any, logger);
+      expect(result).toBe(true);
+      expect(isNftTransactionSpy).toHaveBeenCalledTimes(1);
+    } finally {
+      // Restore original values
+      constants.CREATE_TOKEN_TX_VERSION = originalVersion;
+      process.env = originalEnv;
+    }
+  });
+
+  it('should correctly process a real event from production', () => {
+    expect.hasAssertions();
+
+    // Save original value and mock
+    const originalVersion = constants.CREATE_TOKEN_TX_VERSION;
+    constants.CREATE_TOKEN_TX_VERSION = 2;
+
+    // Set up environment variables
+    const originalEnv = process.env;
+    process.env = { ...originalEnv, NFT_AUTO_REVIEW_ENABLED: 'true' };
+
+    try {
+      // Use the real event data constant
+      const fullNodeData = REAL_NFT_EVENT_DATA;
+
+      // Transform the data using our helper function
+      const txFromEvent = createTransformedEvent(fullNodeData);
+
+      // Verify token handling is correct
+      // First input should be HTR token
+      expect(txFromEvent.inputs[0].token).toBe(hathorLib.constants.HATHOR_TOKEN_CONFIG.uid);
+
+      // First and second outputs should be HTR tokens
+      expect(txFromEvent.outputs[0].token).toBe(hathorLib.constants.HATHOR_TOKEN_CONFIG.uid);
+      expect(txFromEvent.outputs[1].token).toBe(hathorLib.constants.HATHOR_TOKEN_CONFIG.uid);
+
+      // Third output should be the NFT token
+      expect(txFromEvent.outputs[2].token).toBe(fullNodeData.tokens[0]);
+
+      // Check null decoded field is properly handled
+      expect(txFromEvent.outputs[0].decoded).toStrictEqual({});
+
+      // Mock network for validation
+      const mockNetwork = {
+        name: 'testnet',
+      };
+
+      // Test that this transaction is detected as an NFT
+      expect(isNftTransactionSpy).toHaveBeenCalledTimes(0);
+      const shouldInvoke = NftUtils.shouldInvokeNftHandlerForTx(txFromEvent, mockNetwork as any, logger);
```

Comment:
suggestion(non-blocking): Avoid using `any`, as it's prohibited by many standard typescript linting rules.
```suggestion
      const shouldInvoke = NftUtils.shouldInvokeNftHandlerForTx(txFromEvent, mockNetwork as unknown as Network, logger);
```

Reply:
Done, thanks!